core/runtime/wasm_result.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_RUNTIME_WASM_RESULT_HPP
+#define KAGOME_CORE_RUNTIME_WASM_RESULT_HPP
+
+#include "runtime/common.hpp"
+
+namespace kagome::runtime {
+  /**
+   * Result of a call to a Runtime API wasm function is an i64 where first 32
+   * bits are the address and next 32 bits are the size of the returned buffer.
+   */
+  class WasmResult {
```

Comment:
consider making it a struct. 
in case of struct you won't need getters

Reply:
Is it ok for a struct to have constructor and private methods?

Reply:
yes

Reply:
ok, done